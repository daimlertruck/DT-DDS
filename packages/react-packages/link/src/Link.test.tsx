import { withProviders } from '@dt-ui/react-core';
import { render } from '@testing-library/react';

import { Link } from '.';

describe('<Link /> component', () => {
  const ProvidedLink = withProviders(Link);

  it('Should render a html link element with the correct title', () => {
    const { container } = render(
      <ProvidedLink href='/'>Redirect To</ProvidedLink>
    );

    expect(container).toMatchSnapshot();
  });

  it('Should render a disabled html link element', () => {
    const { container } = render(
      <ProvidedLink href='/' isDisabled>
        Redirect To
      </ProvidedLink>
    );

    expect(container).toMatchSnapshot();
  });

  it('Should render an html link element as secondary', () => {
    const { container } = render(
      <ProvidedLink href='/' variant='secondary'>
        Redirect To
      </ProvidedLink>
    );

    expect(container).toMatchSnapshot();
  });
});
