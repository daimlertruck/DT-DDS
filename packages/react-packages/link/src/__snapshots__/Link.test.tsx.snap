// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> component should render a html link element with the correct title 1`] = `
.emotion-0 {
  color: #00677f;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 4px;
  font-family: Daimler CS;
  font-size: 14px;
  font-weight: 700;
  letter-spacing: 0px;
  line-height: 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0:has(span):hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:has(span):hover span {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0:active {
  color: #003946;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:focus-visible {
  outline: 2px solid #000000;
}

<div>
  <a
    class="emotion-0 emotion-1"
    data-testid="link"
    href="/"
  >
    Redirect To
  </a>
</div>
`;

exports[`<Link /> component should render an html link element as secondary 1`] = `
.emotion-0 {
  color: #3e3e3e;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 4px;
  font-family: Daimler CS;
  font-size: 14px;
  font-weight: 700;
  letter-spacing: 0px;
  line-height: 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0:has(span):hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:has(span):hover span {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0:active {
  color: #1d1d1d;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:focus-visible {
  outline: 2px solid #000000;
}

<div>
  <a
    class="emotion-0 emotion-1"
    color="secondary"
    data-testid="link"
    href="/"
  >
    Redirect To
  </a>
</div>
`;
